pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -230,14 +232,64 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 		return configMetaErr
 	}
 
-	if state == clusterMembersChangedUnready {
+	if state == clusterMembersChangedUnready || (cr.Spec.ConfigMetaGenerator != cr.Status.LastConfigMetaGenerator) {
 		cr.Status.SpecGenerationToProcess = &cr.Generation
 	}
 	membersErr := syncMembers(reqLogger, cr, roles, configmetaGen)
 	if membersErr != nil {
 		errLog("members", membersErr)
 		return membersErr
 	}
+	cr.Status.LastConfigMetaGenerator = cr.Spec.ConfigMetaGenerator
+	cr.Status.State = string(clusterReady)
+	// If configMetaGenerator is not incremented then return
```

Comment:
If you want to do all this subsequent work at the point when a cluster transitions from not-ready to ready, you should hook into the code above where we log the "stable" event.

Also I think there's a little redundancy or confusion here stemming from tracking both cluster connections and configmaps with the same generation number. I.e. if one of those changes we still end up doing all the work as if both had changed. That may mean we should track these things with two different generation numbers (or two different hashes as per my other comment).

Reply:
For the first point, we want to do this just after a new config meta is generated for this cluster. If we move this in the block above then after re-generating config meta, we do this in the next run of reconciler and if by then this cluster is updated again then we would lose that.

Reply:
#1 Done. For the other point we need to redesign using hashes. Leaving for next PR